src/EscapeAnalysis.jl

```diff
@@ -1375,6 +1369,22 @@ end
 #     return true
 # end
 
+if isdefined(Core, :arrayfreeze) && isdefined(Core, :arraythaw) &&  isdefined(Core, :mutating_arrayfreeze)
+
+escape_builtin!(::typeof(Core.arrayfreeze), astate::AnalysisState, pc::Int, args::Vector{Any}) =
+    escape_immutable_array!(Array, astate, pc, args)
+escape_builtin!(::typeof(Core.mutating_arrayfreeze), astate::AnalysisState, pc::Int, args::Vector{Any}) =
+    escape_immutable_array!(Array, astate, pc, args)
+escape_builtin!(::typeof(Core.arraythaw), astate::AnalysisState, pc::Int, args::Vector{Any}) =
+    escape_immutable_array!(Core.ImmutableArray, astate, pc, args)
+function escape_immutable_array!(@nospecialize(arytype), astate::AnalysisState, pc::Int, args::Vector{Any})
+    length(args) == 2 || return false
+    argextype(args[2], astate.ir) ⊑ₜ arytype || return false
```

Comment:
With the changes to the `ImmutableArray` builtins in [235ab84](https://github.com/JuliaLang/julia/pull/42465/commits/235ab8443bedf69937d860e7464f90b6f8778538) and [0e5833b](https://github.com/JuliaLang/julia/pull/42465/commits/0e5833bea0a6cd40eb3d81a3b840564c7ee85e8d), this looks straightforward to me :+1:

Reply:
yes!